sale_loyalty_partner_applicability/models/sale_order.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import ast
+
+from odoo import _, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _is_valid_partner(self, program):
+        is_valid = False
+        for rule in program.rule_ids:
+            if not is_valid and rule.rule_partners_domain != "[]":
+                domain = ast.literal_eval(rule.rule_partners_domain) + [
+                    ("id", "=", self.partner_id.id)
```

Comment:
Add a hook method to get the partner domain (just return this one for the moment) so we can use it for the migration of `coupon_commercial_partner_applicability`